[WIP] arm64: dts: qcom: coolpad-coolpady80d: initial support

nphuracm opened the pull request:
This is a very early and dirty attempt in mainlining the device coolpad-coolpady80d.
* UART is known to work. (Personal information: I lost UART output.)
* **USB networking does NOT work for now.**
* Could not verify if panel is working due to the panel not working from the primary bootloader (lk1st) and boot seeming to fail at an early stage where panel is not yet initialized by the kernel.
* The DTS is **not yet compiling.**

nphuracm committed:
Preliminary support for coolpad-coolpady80d (not yet compiling)

nphuracm committed:
duh why

nphuracm committed:
attempt at supporting coolpad-coolpady80d

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8916-coolpad-coolpady80d.dts:
```diff
@@ -0,0 +1,228 @@
+/dts-v1/;
+
+#include "msm8916-pm8916.dtsi"
+#include "msm8916.dtsi"
+#include "pm8916.dtsi"
```
```suggestion
```

They are already included from `msm8916-pm8916.dtsi`

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8916-coolpad-coolpady80d.dts:
```diff
@@ -0,0 +1,228 @@
+                compatible = "mdss,otm9605a-lide-550";
+                reg = <0>;
+
+                reset-gpios = <78 25 0>;
```
```suggestion
                reset-gpios = <&msmgpio 25 GPIO_ACTIVE_HIGH>;
```

78?

`GPIO_ACTIVE_HIGH` defined as `0`. See `dt-bindings/gpio/gpio.h`

wonderfulShrineMaidenOfParadise commented on drivers/gpu/drm/panel/Makefile:
```diff
@@ -39,6 +39,7 @@ obj-$(CONFIG_DRM_PANEL_MANTIX_MLAF057WE51) += panel-mantix-mlaf057we51.o
 obj-$(CONFIG_DRM_PANEL_OLIMEX_LCD_OLINUXINO) += panel-olimex-lcd-olinuxino.o
 obj-$(CONFIG_DRM_PANEL_ORISETECH_OTM8009A) += panel-orisetech-otm8009a.o
 obj-$(CONFIG_DRM_PANEL_OSD_OSD101T2587_53TS) += panel-osd-osd101t2587-53ts.o
+obj-$(CONFIG_DRM_PANEL_OTM9605A_LIDE_550) += panel-otm9605a-lide-550.o
```
Generate it with https://github.com/msm8916-mainline/linux-panel-drivers, a wrapper of lmdpdg instead.
Then you just need `DRM_PANEL_MSM8916_GENERATED`, which is enabled by default.

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8916-coolpad-coolpady80d.dts:
```diff
@@ -0,0 +1,228 @@
+	status = "okay";
+};
+/*
+&msmgpio {
```
Move this to the bottom, after `&smd_rpm_regulators`

wonderfulShrineMaidenOfParadise replied:
```suggestion
&msmgpio {
	mdss_default: mdss-default-state {
		pins = "gpio25";
		function = "gpio";

		drive-strength = <8>;
		bias-disable;
	};

	mdss_sleep: mdss-sleep-state {
		pins = "gpio25";
		function = "gpio";

		drive-strength = <2>;
		bias-pull-down;
	};
```

Pinctrl for mdss is missing.

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8916-coolpad-coolpady80d.dts:
```diff
@@ -0,0 +1,228 @@
+
+//generated by lmdpdg
+&dsi0 {
+        panel@0 {
```
It seems that you also need backlight-gpio and a fixed-regulator. Check your downstream dts and schematics for details.

nphuracm replied:
I have this attribute in downstream:
```
qcom,mdss_dsi@1a98000 {
 /* ... */
qcom,platform-bklight-en-gpio = <0x4e 0x62 0x00>;
 /* ... */
};
```
So that translates to:
```
      panel@0 {
                compatible = "coolpad,otm9605a-lide";
                reg = <0>;

                reset-gpios = <&msmgpio 25 GPIO_ACTIVE_HIGH>;
                backlight-gpio = <&msmgpio 98 GPIO_ACTIVE_HIGH>;
                
                port {
                        panel_in: endpoint {
                                remote-endpoint = <&dsi0_out>;
                        };
                };
        };
```
Right?

wonderfulShrineMaidenOfParadise replied:
Looks right. Try [fixed-regulator](https://www.kernel.org/doc/Documentation/devicetree/bindings/regulator/fixed-regulator.txt) next.

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8916-coolpad-coolpady80d.dts:
```diff
@@ -0,0 +1,228 @@
+};
+
+//generated by lmdpdg
+&dsi0 {
```
```suggestion
&dsi0 {
	pinctrl-0 = <&mdss_default>;
	pinctrl-1 = <&mdss_sleep>;
	pinctrl-names = "default", "sleep";

```

Pinctrl is missing.

nphuracm committed:
Update arch/arm64/boot/dts/qcom/msm8916-coolpad-coolpady80d.dts

Co-authored-by: Raymond Hackley <raymondhackley@protonmail.com>

nphuracm committed:
Update arch/arm64/boot/dts/qcom/msm8916-coolpad-coolpady80d.dts

Co-authored-by: Raymond Hackley <raymondhackley@protonmail.com>

nphuracm committed:
Update arch/arm64/boot/dts/qcom/msm8916-coolpad-coolpady80d.dts

Co-authored-by: Raymond Hackley <raymondhackley@protonmail.com>

nphuracm committed:
Update arch/arm64/boot/dts/qcom/msm8916-coolpad-coolpady80d.dts

Co-authored-by: Raymond Hackley <raymondhackley@protonmail.com>

nphuracm committed:
Update arch/arm64/boot/dts/qcom/msm8916-coolpad-coolpady80d.dts

Co-authored-by: Raymond Hackley <raymondhackley@protonmail.com>

nphuracm committed:
Update arch/arm64/boot/dts/qcom/msm8916-coolpad-coolpady80d.dts

Co-authored-by: Raymond Hackley <raymondhackley@protonmail.com>

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8916-coolpad-coolpady80d.dts:
```diff
@@ -0,0 +1,264 @@
+	pinctrl-1 = <&mdss_sleep>;
+	pinctrl-names = "default", "sleep";
+
+	pinctrl-0 = <&mdss_default>;
+	pinctrl-1 = <&mdss_sleep>;
+	pinctrl-names = "default", "sleep";
+
```
Duplicated?

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8916-coolpad-coolpady80d.dts:
```diff
@@ -0,0 +1,264 @@
+		drive-strength = <2>;
+		bias-pull-down;
+	};
+	mdss_default: mdss-default-state {
+		pins = "gpio25";
+		function = "gpio";
+
+		drive-strength = <8>;
+		bias-disable;
+	};
+
+	mdss_sleep: mdss-sleep-state {
+		pins = "gpio25";
+		function = "gpio";
+
+		drive-strength = <2>;
+		bias-pull-down;
+	};
```
Ditto.

wonderfulShrineMaidenOfParadise commented on drivers/gpu/drm/panel/Kconfig:
```diff
@@ -736,5 +736,6 @@ config DRM_PANEL_XINPENG_XPP055C272
 	  Say Y here if you want to enable support for the Xinpeng
 	  XPP055C272 controller for 720x1280 LCD panels with MIPI/RGB/SPI
 	  system interfaces.
+
```
```suggestion
```

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8916-coolpad-coolpady80d.dts:
```diff
@@ -0,0 +1,264 @@
+	pinctrl-names = "default", "sleep";
+
+        panel@0 {
+                compatible = "mdss,otm9605a-lide-550";
```
```suggestion
                compatible = "coolpad,otm9605a-lide";
```

Or `coolpad,coolpady80d-panel`, which is added to your lk2nd/lk1st commit.

nphuracm committed:
Update drivers/gpu/drm/panel/Kconfig

Co-authored-by: Raymond Hackley <raymondhackley@protonmail.com>

nphuracm commented:
Now there's a problem.
```
arch/arm64/boot/dts/qcom/msm8916-coolpad-coolpady80d.dts:22.12-35.4: ERROR (phandle_references): /gpio-keys: Reference to non-existent node or label "gpio_keys_default"
```

nphuracm committed:
fix missing pinctrl.

nphuracm committed:
add iris config

nphuracm committed:
ah fuck

Commit:
squash! MSM8916: drm/panel: Generate using linux-mdss-dsi-panel-driver-generator

X-Code-Generator: msm8916-mainline/linux-panel-drivers@b891186
Signed-off-by: lmdpdg <lmdpdg@localhost>

nphuracm committed:
some fixes

nphuracm committed:
oopsie

nphuracm committed:
Will it really work...?

nphuracm committed:
Things work now.
USB networking works.
eMMC works.
Directly copying some files also make WCNSS work.

Commit:
squash! MSM8916: drm/panel: Generate using linux-mdss-dsi-panel-driver-generator

X-Code-Generator: msm8916-mainline/linux-panel-drivers@b891186
Signed-off-by: lmdpdg <lmdpdg@localhost>

Commit:
squash! MSM8916: drm/panel: Generate using linux-mdss-dsi-panel-driver-generator

X-Code-Generator: msm8916-mainline/linux-panel-drivers@b891186
Signed-off-by: lmdpdg <lmdpdg@localhost>

nphuracm committed:
attempt in enabling panel regulator - TPS65132

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8916-coolpad-coolpady80d.dts:
```diff
@@ -33,6 +33,30 @@
+			regulator-name = "outn";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			enable-gpios = <&pm8916_gpios 97 GPIO_ACTIVE_HIGH>;
```
It don't look right when pin 97 is referenced twice. Also do you mean `&msmgpio` instead of `&pm8916_gpio`?

wonderfulShrineMaidenOfParadise commented:
Alternatively you can forget `ti,tps65132`. See how the panel on harpia is done in 581dbf9117a7a116b19ee533ebb27bae1a20dddf.

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8916-coolpad-coolpady80d.dts:
```diff
@@ -0,0 +1,290 @@
+
+		label = "GPIO Buttons";
+
+		volume-up {
```
```suggestion
		button-volume-up {
```

Ref: b08f5cbd69dcd25f5ab2a0798fe3836a97a9d7c6

nphuracm committed:
Update arch/arm64/boot/dts/qcom/msm8916-coolpad-coolpady80d.dts

Co-authored-by: Raymond Hackley <raymondhackley@protonmail.com>

nphuracm committed:
?

nphuracm committed:
Some improvements on regulator declaration and fix volume up key
reference

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8916-coolpad-coolpady80d.dts:
```diff
@@ -0,0 +1,290 @@
+&pm8916_gpios {
+	disp_en_default: disp-en-default-state {
+		pins = "gpio97";
+		function = PMIC_GPIO_FUNC_NORMAL;
```
You don't have `PMIC_GPIO_FUNC_NORMAL` defined and there is nowhere to be found in your dts/dtsi.

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8916-coolpad-coolpady80d.dts:
```diff
@@ -0,0 +1,290 @@
+};
+
+//TODO: Verify GPIOs!
+&pm8916_gpios {
+	disp_en_default: disp-en-default-state {
+		pins = "gpio97";
+		function = PMIC_GPIO_FUNC_NORMAL;
+		power-source = <PM8916_GPIO_L2>;
+		bias-disable;
+	};
+};
```
```suggestion
&msmgpio {
	disp_en_default: disp-en-default-state {
		pins = "gpio97";
		function = "gpio";
		drive-strength = <2>;
		bias-disable;
	};
};
```

https://github.com/msm8916-mainline/linux/commit/581dbf9117a7a116b19ee533ebb27bae1a20dddf#diff-f55185053afe3696071b66ada5f4a4bf0ec46466661dd8164f4e0f0ea6c20881R325-R337

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8916-coolpad-coolpady80d.dts:
```diff
@@ -0,0 +1,290 @@
+			regulator-name = "outp";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			enable-gpios = <&pm8916_gpios 97 GPIO_ACTIVE_HIGH>;
```
```suggestion
			enable-gpios = <&msmgpio 97 GPIO_ACTIVE_HIGH>;
```

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8916-coolpad-coolpady80d.dts:
```diff
@@ -0,0 +1,290 @@
+		};
+	};
+	
+	tps65132@3e {
```
Mostly it should be on an i2c bus

nphuracm committed:
SOME CHANGESc

nphuracm committed:
come onnnnnn

nphuracm committed:
guess i'll keep typing nonsense here before i finally figure something
out

nphuracm committed:
fuuuuck

nphuracm committed:
Merge branch 'msm8916/6.2-rc5' of https://github.com/nphuracm/msm8916-mainline-linux into msm8916/6.2-rc5

Commit:
squash! MSM8916: drm/panel: Generate using linux-mdss-dsi-panel-driver-generator

X-Code-Generator: msm8916-mainline/linux-panel-drivers@b891186
Signed-off-by: lmdpdg <lmdpdg@localhost>

Commit:
squash! MSM8916: drm/panel: Generate using linux-mdss-dsi-panel-driver-generator

X-Code-Generator: msm8916-mainline/linux-panel-drivers@b891186
Signed-off-by: lmdpdg <lmdpdg@localhost>

Commit:
squash! MSM8916: drm/panel: Generate using linux-mdss-dsi-panel-driver-generator

X-Code-Generator: msm8916-mainline/linux-panel-drivers@b891186
Signed-off-by: lmdpdg <lmdpdg@localhost>

Commit:
squash! MSM8916: drm/panel: Generate using linux-mdss-dsi-panel-driver-generator

X-Code-Generator: msm8916-mainline/linux-panel-drivers@b891186
Signed-off-by: lmdpdg <lmdpdg@localhost>

nphuracm committed:
?????

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8916-coolpad-coolpady80d.dts:
```diff
@@ -0,0 +1,303 @@
+	 * Unfortunately, the schematics were incomplete, and we are not
+	 * able to locate that pin right away.
+	 */
+	tps65132@3e {
```
```suggestion
	regulator {
```

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8916-coolpad-coolpady80d.dts:
```diff
@@ -0,0 +1,303 @@
+	 */
+	tps65132@3e {
+		compatible = "fixed-regulator";
+		regulator-name = "fixed-supply";
+		regulator-min-microvolt = "5000000";
+		regulator-max-microvolt = "5000000";
```
```suggestion
		regulator-name = "fixed-supply";
		regulator-min-microvolt = "5000000";
		regulator-max-microvolt = "5000000";
		regulator-always-on;
```
To test a regulator, set the bool `regulator-always-on;`.

nphuracm committed:
Update arch/arm64/boot/dts/qcom/msm8916-coolpad-coolpady80d.dts

Co-authored-by: Raymond Hackley <raymondhackley@protonmail.com>

nphuracm committed:
Update arch/arm64/boot/dts/qcom/msm8916-coolpad-coolpady80d.dts

Co-authored-by: Raymond Hackley <raymondhackley@protonmail.com>

nphuracm committed:
Merge branch 'msm8916/6.2-rc5' of https://github.com/nphuracm/msm8916-mainline-linux into msm8916/6.2-rc5

Commit:
squash! MSM8916: drm/panel: Generate using linux-mdss-dsi-panel-driver-generator

X-Code-Generator: msm8916-mainline/linux-panel-drivers@b891186
Signed-off-by: lmdpdg <lmdpdg@localhost>

nphuracm committed:
duhhhh

Commit:
squash! MSM8916: drm/panel: Generate using linux-mdss-dsi-panel-driver-generator

X-Code-Generator: msm8916-mainline/linux-panel-drivers@b891186
Signed-off-by: lmdpdg <lmdpdg@localhost>

Commit:
squash! MSM8916: drm/panel: Generate using linux-mdss-dsi-panel-driver-generator

X-Code-Generator: msm8916-mainline/linux-panel-drivers@b891186
Signed-off-by: lmdpdg <lmdpdg@localhost>

Commit:
squash! MSM8916: drm/panel: Generate using linux-mdss-dsi-panel-driver-generator

X-Code-Generator: msm8916-mainline/linux-panel-drivers@b891186
Signed-off-by: lmdpdg <lmdpdg@localhost>

nphuracm committed:
wait a minute

nphuracm committed:
Again

Commit:
squash! MSM8916: drm/panel: Generate using linux-mdss-dsi-panel-driver-generator

X-Code-Generator: msm8916-mainline/linux-panel-drivers@b891186
Signed-off-by: lmdpdg <lmdpdg@localhost>

nphuracm committed:
ugh

Commit:
squash! MSM8916: drm/panel: Generate using linux-mdss-dsi-panel-driver-generator

X-Code-Generator: msm8916-mainline/linux-panel-drivers@b891186
Signed-off-by: lmdpdg <lmdpdg@localhost>

Commit:
squash! MSM8916: drm/panel: Generate using linux-mdss-dsi-panel-driver-generator

X-Code-Generator: msm8916-mainline/linux-panel-drivers@b891186
Signed-off-by: lmdpdg <lmdpdg@localhost>

Commit:
squash! MSM8916: drm/panel: Generate using linux-mdss-dsi-panel-driver-generator

X-Code-Generator: msm8916-mainline/linux-panel-drivers@b891186
Signed-off-by: lmdpdg <lmdpdg@localhost>

Commit:
squash! MSM8916: drm/panel: Generate using linux-mdss-dsi-panel-driver-generator

X-Code-Generator: msm8916-mainline/linux-panel-drivers@b891186
Signed-off-by: lmdpdg <lmdpdg@localhost>

nphuracm committed:
uuuuuuuuuuuuhhhhhhhhhhhhhhh

nphuracm committed:
deeming myself stupid over and over again

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8916-coolpad-coolpady80d.dts:
```diff
@@ -0,0 +1,329 @@
+	status = "okay";
+};
+
+&pronto {
+	status = "okay";
+	//this device uses WCN3620
+};
```
A note about new changes since 6.3+: 3244442406ff49e8f75a1f2def211c497710570f

nphuracm committed:
arm64: dts: qcom: msm8916: coolpad-coolpady80d: Attempt to get display

nphuracm committed:
this will be changed

Commit:
squash! MSM8916: drm/panel: Generate using linux-mdss-dsi-panel-driver-generator

X-Code-Generator: msm8916-mainline/linux-panel-drivers@b891186
Signed-off-by: lmdpdg <lmdpdg@localhost>

Commit:
squash! MSM8916: drm/panel: Generate using linux-mdss-dsi-panel-driver-generator

X-Code-Generator: msm8916-mainline/linux-panel-drivers@b891186
Signed-off-by: lmdpdg <lmdpdg@localhost>

Commit:
squash! MSM8916: drm/panel: Generate using linux-mdss-dsi-panel-driver-generator

X-Code-Generator: msm8916-mainline/linux-panel-drivers@b891186
Signed-off-by: lmdpdg <lmdpdg@localhost>

Commit:
squash! MSM8916: drm/panel: Generate using linux-mdss-dsi-panel-driver-generator

X-Code-Generator: msm8916-mainline/linux-panel-drivers@b891186
Signed-off-by: lmdpdg <lmdpdg@localhost>

nphuracm committed:
1

Signed-off-by: Chen Junze <cjz20070630@yeah.net>

nphuracm commented:
This has been scrapped for a looooong time. Since many breaking changes has happened during this period, and some serious condition with my current test device, without much opportunity for me to focus much on this, I'll be closing this PR and redo everything at a future time.
